code/modules/reagents/reagents.dm

```diff
@@ -49,9 +49,52 @@
 	var/heating_sound = 'sound/effects/bubbles.ogg'
 	var/list/heating_products
 
+	var/constantMetabolism = FALSE	// if metabolism factor should not change with volume or blood circulation
+
+	var/NSA = 10 // Nerve system accumulations
+
 /datum/reagent/proc/remove_self(amount) // Shortcut
 	holder.remove_reagent(id, amount)
 
+/datum/reagent/proc/consumed_amount(mob/living/carbon/M, var/alien, var/location)
+	var/removed = metabolism
+	if(ingest_met && (location == CHEM_INGEST))
+		removed = ingest_met
+	if(touch_met && (location == CHEM_TOUCH))
+		removed = touch_met
+	// on half of overdose, chemicals will start be metabolized faster, 
+	// also blood circulation affects chemical strength (meaining if target has low blood volume or has something that lowers blood circulation chemicals will be consumed less and effect will diminished)
+	if(location == CHEM_BLOOD)
+		if(!constantMetabolism)
+			if(overdose)
+				removed = CLAMP(metabolism * volume/(overdose/2) * M.get_blood_circulation()/100, metabolism * REAGENTS_MIN_EFFECT_MULTIPLIER, metabolism * REAGENTS_MAX_EFFECT_MULTIPLIER)
+			else
+				removed = CLAMP(metabolism * volume/(REAGENTS_OVERDOSE/2) * M.get_blood_circulation()/100, metabolism * REAGENTS_MIN_EFFECT_MULTIPLIER, metabolism * REAGENTS_MAX_EFFECT_MULTIPLIER)
+	removed = round(removed, 0.01)
+	removed = min(removed, volume)
+	if(M.name == "Jhony Mccune")
+		world << "removed [name] = [removed]"
+	return removed
+
+// "Removed" to multiplier
+// will return multiplier of how much value is more or less than default metabolism amount
+// all chem effects should be multiplied by return of this proc
+/datum/reagent/proc/RTM(var/removed, var/location)
```

Comment:
All caps proc names are also a little weird. I don't know what the precedent on this is though so if the other maintainers are cool with all caps proc and var names I'm not gonna fight them on it.

Reply:
i've changed other occurrences but intend to leave this one, since in feels like defines in some way.